File: csharp/ScratchCard.cs
// csharp/ScratchCard.cs
namespace ECommerce
{
    public class ScratchCard : Product
    {
        public ScratchCard(string name, double price, int quantity)
            : base(name, price, quantity)
        {
        }

        public override bool IsExpired()
        {
            return false;
        }

        public override bool RequiresShipping()
        {
            return false;
        }
    }
}

File: cpp/ScratchCard.cpp
#include "ScratchCard.h"

ScratchCard::ScratchCard(std::string name, double price, int quantity)
    : Product(name, price, quantity) {}

bool ScratchCard::isExpired() const {
    return false;
}

bool ScratchCard::requiresShipping() const {
    return false;
}
